lib/matplotlib/cbook/__init__.py

```diff
@@ -1730,12 +1737,16 @@ def _safe_first_non_none(obj, skip_none=True):
                                "as input")
         return next(iter(obj))
     elif isinstance(obj, np.flatiter):
+        # TODO do the finite filtering on this
         return obj[0]
     elif isinstance(obj, collections.abc.Iterator):
         raise RuntimeError("matplotlib does not "
                            "support generators as input")
     else:
-        return next(val for val in obj if val is not None)
+        return next(
+            val for val in obj
+            if val is not None and safe_isfinite(val)
```

Comment:
Logically, should we absorb the None check in safe_isfinite? -I think None is not finite.

Reply:
I think we need to check this here:

```
In [62]: np.isfinite(None)
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
Cell In [63], line 1
----> 1 np.isfinite(None)

TypeError: ufunc 'isfinite' not supported for the input types, and the inputs could not be safely coerced to any supported types according to the casting rule ''safe''
```

Reply:
Yes, we need the check. I was suggesting that it could logically belong inside `safe_isfinite` (by the  argument that `None` is not a finite number, similar to NaN).

i.e.
```
def safe_isfinite(val):
    if val is None:
        return False
    ...
```

and here `next(val for val in obj if safe_isfinite(val))`

I just realized that this is an inner function, so it's not too important, but I'd still have a slight preference. I'll approve and you can decide whether you want to change it or leave it.

Reply:
I agree that is nicer.